kie-server-parent/kie-server-api/pom.xml

```diff
@@ -24,10 +24,17 @@
       <artifactId>kie-api</artifactId>
       <!-- scope>provided</scope -->
     </dependency>
+
+    <!-- we need to find a way to remove the following 2 dependencies from this jar -->
     <dependency>
       <groupId>org.drools</groupId>
       <artifactId>drools-core</artifactId>
-      <!-- scope>provided</scope -->
+      <scope>optional</scope>
```

Comment:
optional is not a scope, should be
<optional>true</optional>
instead
I am preparing a PR for edson's fork.